icu4c/source/i18n/messageformat2_parser.h

```diff
@@ -84,15 +105,55 @@ namespace message2 {
 	    UChar   postContext[U_PARSE_CONTEXT_LEN];
 	} MessageParseError;
 
-	Parser(const UnicodeString &input, MFDataModel::Builder& dataModelBuilder, StaticErrors& e, UnicodeString& normalizedInputRef)
-	  : source(input), index(0), errors(e), normalizedInput(normalizedInputRef), dataModel(dataModelBuilder) {
+	Parser(const UnicodeString &input,
+               MFDataModel::Builder& dataModelBuilder,
+               StaticErrors& e,
+               UnicodeString& normalizedInputRef,
+               UErrorCode& status)
+	  : contentChars(unisets::get(unisets::CONTENT)),
+            whitespaceChars(unisets::get(unisets::WHITESPACE)),
+            bidiControlChars(unisets::get(unisets::BIDI)),
+            alphaChars(unisets::get(unisets::ALPHA)),
+            digitChars(unisets::get(unisets::DIGIT)),
+            nameStartChars(unisets::get(unisets::NAME_START)),
+            nameChars(unisets::get(unisets::NAME_CHAR)),
+            textChars(unisets::get(unisets::TEXT)),
+            quotedChars(unisets::get(unisets::QUOTED)),
+            escapableChars(unisets::get(unisets::ESCAPABLE)),
+            source(input), index(0), errors(e), normalizedInput(normalizedInputRef), dataModel(dataModelBuilder) {
+            (void) status;
 	  parseError.line = 0;
 	  parseError.offset = 0;
 	  parseError.lengthBeforeCurrentLine = 0;
 	  parseError.preContext[0] = '\0';
 	  parseError.postContext[0] = '\0';
 	}
 
+        UnicodeSet initContentChars(UErrorCode& status);
```

Comment:
are the return type of these initX method "UnicodeSet" ? or it should be "UnicodeSet* "?

Reply:
Should be `UnicodeSet*`, but these declarations aren't actually necessary as the functions are defined before they're used.